cfgov/login/password_validation.py

```diff
@@ -0,0 +1,79 @@
+import functools
+import re
+from datetime import timedelta
+
+from django.contrib.auth import hashers
+from django.core.exceptions import ValidationError
+from django.utils import timezone
+
+
+class PasswordRegexValidator:
+    """Validate that the password matches a given regex"""
+
+    regex = ""
+    message = "Enter a valid password"
+
+    def __init__(self, regex=None, message=None):
+        if regex is not None:
+            self.regex = re.compile(regex)
+        if message is not None:
+            self.message = message
+
+    def validate(self, password, user=None):
+        if not self.regex.search(str(password)):
+            raise ValidationError(self.message)
+
+    def get_help_text(self):
+        return self.message
+
+
+class AgeValidator:
+    """Validate that the password hasn't been changed too recently"""
+
+    def __init__(self, hours):
+        self.hours = hours
+
+    def validate(self, password, user=None):
+        if user is None:  # pragma: no cover
+            return
+
+        # Superusers can always change their password.
+        if user.is_superuser:
+            return
+
```

Comment:
```suggestion
        # Superusers can always change their password.
        if user.is_superuser:
            return

```

I'm having second thoughts about introducing this in my PR. I think I misinterpreted [this logic](https://github.com/cfpb/consumerfinance.gov/blob/7b6190d4c43c8fcffb5a126e1285ecb1e731344d/cfgov/login/signals.py#L28-L30) in the existing code: the current logic allows new superusers to keep using their initial password, it doesn't remove the restriction on re-changing changed passwords down the line.

Reply:
Well, with SSO, all of this has to go away anyway 😇

Reply:
Done in [df3aac2](https://github.com/cfpb/consumerfinance.gov/pull/7650/commits/df3aac2c721885984228ccfe51fcc7c4804ddb29)